test/on.js

```diff
@@ -0,0 +1,15 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+
+describe('on', function() {
+  it('compares identity using identity', function() {
+    const eqBy = R.on((a, b) => a === b);
+    eq(eqBy(R.identity, R.identity, R.identity), true);
+  });
+
+  it('finds a substring using lower case', function() {
+    const containsInsensitive = R.on(R.includes, R.toLower);
+    eq(containsInsensitive('o', 'FOO'), true);
+  });
+});
```

Comment:
This test, on the other hand is precisely what's needed to demonstrate the behavior of the function.

But the test description is still off.  These BDD-style test descriptions are meant to be read as sentences.  For instance, these are the test results for `filter`: 

```
    filter
    ✓ reduces an array to those matching a filter
    ✓ returns an empty array if no element matches
    ✓ returns an empty array if asked to filter an empty array
    ✓ filters objects
    ✓ dispatches to passed-in non-Array object with a `filter` method
```

They describe what `filter` does, not what the test does.  The test should simply be a demonstration of the that description.  Here is what `on` looks like:

```
  on
    ✓ compares identity using identity
    ✓ finds a substring using lower case
 ```

It describes the tests, not the function.

Perhaps something like this for the description?:

```
  on
    ✓ transforms two values with a common function and supplies their results to a combining function
 ```

It is a bit of a mouthful, but this explicitly describes the function.  This might also work, but is even longer: "combines two values by calling a common transformation function on each and a combining function on the results".

Reply:
How about

>combines two values using a transformer function on each value

Or, perhaps more tersely:

>combines and transforms two values

?

Reply:
Sure, I like the first one a lot, but either would do.

Reply:
I think it's fine.  The test reports are meant to be readable, but they are not the main documentation.  I just like the idea that they can serve as secondary documentation.  I personally often look first to the unit test suite when trying to understand a new system.